boa/src/value/mod.rs

```diff
@@ -425,8 +425,9 @@ impl JsValue {
         K: Into<PropertyKey>,
     {
         let _timer = BoaProfiler::global().start_event("Value::has_field", "value");
+        // todo: call `__has_property__` instead of directly getting from object
```

Comment:
Why can't this be called?

Reply:
I think that's authored to me because I rearranged the internal methods and object operations. I'll check if I can solve the todos.

Reply:
Ok, this would have to be its own PR. There are a lot of changes to be made in the AST and the `environment` module if we want to call `[[HasProperty]]` directly.